core-bundle/tests/Crawl/Escargot/Subscriber/SearchIndexSubscriberTest.php

```diff
@@ -498,10 +497,7 @@ static function (string $key) use ($statusCode) {
         return $response;
     }
 
-    /**
-     * @return TranslatorInterface&MockObject
-     */
-    private function getTranslator(): TranslatorInterface
+    private function getTranslator()
```

Comment:
Is it necessary to remove the `TranslatorInterface` return type so that the static analysis tools can infer `TranslatorInterface&MockObject` as the correct type?

Reply:
> Is it necessary to remove the TranslatorInterface return type so that the static analysis tools can infer TranslatorInterface&MockObject as the correct type?

If so I'd vote to keep the annotation instead.

Maybe this change would also make it harder to switch to real PHP8.1 intersection types at some point in a semi-automated way.

Reply:
I have added the return types to the mock methods again in c1307c6ab1b38e929e45b95fffa1bcf594ba86b0. I have also fixed the methods which did not follow the naming rule "methods that return a mock object must begin with `mock`".